### ISSUE TITLE
Add default bounding box for compound models

### ISSUE THREAD

**Issue by lpsinger** [astropy/astropy#16951]
### What is the problem this feature will solve?

There is a common need to be able to integrate compound models (e.g., synphot). Much of the complex machinery to enable this relies on guessing the limits of integration.

Many models self-document the domain over which they have significant values using the `bounding_box` property. The exception is compound models, which don't provide a useful default bounding box implementation.

### Describe the desired outcome

Implement a default bounding box method for compound models that determines the bounding box by traversing the model tree (recursively) and combining intervals appropriately.

### Additional context

According to the [documentation](https://docs.astropy.org/en/stable/modeling/models.html#efficient-model-rendering-with-bounding-boxes):

> All [Model](https://docs.astropy.org/en/stable/api/astropy.modeling.Model.html#astropy.modeling.Model) subclasses have a [bounding_box](https://docs.astropy.org/en/stable/api/astropy.modeling.Model.html#astropy.modeling.Model.bounding_box) attribute that can be used to set the limits over which the model is significant. This greatly improves the efficiency of evaluation when the input range is much larger than the characteristic width of the model itself.

I read "significant" as "significantly different from zero." If that is the case, then you can define some simple behavior for compound model operations:

| Op | Bounds |
| - | - |
| * | intersection |
| +, - | union |
| / | LHS if RHS is unbounded, else undefined |

etc.

**Comment by pllim** [comment]
Would something similar to this help?

https://github.com/spacetelescope/synphot_refactor/blob/7034d1ac53401179ceeb06da799228ab7f60c95d/synphot/models.py#L793

**Comment by lpsinger** [comment]
Yes, that's the idea, but I'm looking for native support in Astropy modeling.
